### ISSUE TITLE
[HOLD for payment 2024-01-11] [$500] [Wave 6: Workspace Chats] Submit button on expense view should not be green Collect workspaces

### ISSUE THREAD

**Issue by Christinadobrzyn** [Expensify/App#33443]
Related Slack chat: https://expensify.slack.com/archives/C02MW39LT9N/p1703187255250009?thread_ts=1703187017.215259&cid=C02MW39LT9N

## Action Performed:

Pre-testing requirement: Collect workspace type with the isPolicyExpenseChatEnabled NVP set ([SO](https://stackoverflowteams.com/c/expensify/questions/17518)).

Note: Scheduled submit should be set to any frequency other than Manually

1. Sign-in for the first time using the member account
• Verify that you land in the workspace chat
3. Go to global create (`+`) > Request money > enter an amount > confirm
4. Tap the report preview component to navigate to the expense report view
• Verify that a banner is in the page header with the label `Next steps` :white_check_mark:
• Verify that the next steps text matches the scheduled submit cadence you set on the paid policy. E.g “These expenses are scheduled to submit on Sunday. No action required.” :white_check_mark:
• Verify that a dark blue `Submit` button is in the page header beneath the “Next steps” banner :x:

## Expected Result:

Submit button in the expense report view should be the standard dark blue button colour - indicating it can't be actioned

## Actual Result:

Submit button in the expense report view is also green indicating it can be actioned

<img width="903" alt="image" src="https://github.com/Expensify/App/assets/51066321/0eada0b7-41dd-45af-ac54-2759e02f218c">

POLISH: 

After hitting submit the next steps banner disappears and takes a while to reappear with the correct message.

Action step:

1. Tap the `Submit` button
• Verify the text in the next steps banner updates to reflect the report has been submitted and is awaiting approval: “Waiting for %approverName% to review these requests” :x:
• Verify a system message is added to the expense report from the submitter that reads: “submitted %reportTotal%” :white_check_mark:

https://github.com/Expensify/App/assets/51066321/dd9a169d-0cfa-4630-954a-bdef93d35914

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0194651512192b0478</li>
        <li>Upwork Job ID: 1737950405980835840</li>
        <li>Last Price Increase: 2023-12-21</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @joekaufmanexpensify</details>

**Comment by joekaufmanexpensify** [comment]
I can reproduce. A few notes:

>Submit button in the expense report view should be the standard dark blue button colour - indicating it can't be actioned

For the expected behavior, based on [the design doc](https://docs.google.com/document/d/1vZtGpW_Fy-Wq_R6gvnjdmWVA86FnR42y3LtnNA--vcY/edit?pli=1#bookmark=id.qfa9x8cv1rod), it seems like the `Submit` button can just be actioned. It's just supposed to have the lighter button color to indicate that no action is required, unless someone wants to submit early. But LMK if I am not correct @trjExpensify @Christinadobrzyn @greg-schroeder. 

>Tap the Submit button
• Verify the text in the next steps banner updates to reflect the report has been submitted and is awaiting approval: “Waiting for %approverName% to review these requests” ❌

For this, I can get the banner to update, but it's only after navigating from the report level review to the workspace chat, and then clicking back into the report. I think we'd want this banner to "live update" though, so makes sense to me to change this.

**Comment by joekaufmanexpensify** [comment]
Also going to make this weekly since I'm OOO for the holidays. Will adjust prioritization when I'm back. @alitoshmatov if you're working feel free to review/sign off on a proposal if you're around!

**Comment by mountiny** [comment]
Thanks for the issue, I have been working on this in another issue so I will take it on

I have added the `isHarvestingEnabled` key to be passed from the backend and we need to use that key to know of the scheduled submit is on or not

**Comment by mountiny** [comment]
Sorry @abzokhattab This PR was taken internal

**Comment by mountiny** [comment]
Not overdue

**Comment by mountiny** [comment]
PR is in review

**Comment by mountiny** [comment]
Assigning @shubham1206agra as C+ given they have volunteered to reviw tihs Pr which requires certain set up

**Comment by joekaufmanexpensify** [comment]
PR merged

**Comment by mountiny** [comment]
$500 to @shubham1206agra

**Comment by joekaufmanexpensify** [comment]
@shubham1206agra could you please handle the BZ checklist so we can prep to issue payment this week?

**Comment by mountiny** [comment]
@joekaufmanexpensify I think we can skip this one around, it is still working on a new feature so I dont think there is a useful regression to point out in this case and regression tests will be added as part of larger block of testing collect submitters

**Comment by joekaufmanexpensify** [comment]
Got it. Sounds good! I'll just issue the $500 payment to @shubham1206agra once the 7 day regression period is up.

**Comment by joekaufmanexpensify** [comment]
@shubham1206agra offer sent for $500!

**Comment by shubham1206agra** [comment]
Accepted

**Comment by joekaufmanexpensify** [comment]
Great. TY!

**Comment by joekaufmanexpensify** [comment]
All set to issue payment

**Comment by joekaufmanexpensify** [comment]
All set.
